app/src/main/java/net/aiscope/gdd_app/network/ModelDtoExt.kt

```diff
@@ -0,0 +1,39 @@
+package net.aiscope.gdd_app.network
+
+import android.os.Build
+import net.aiscope.gdd_app.BuildConfig
+import net.aiscope.gdd_app.model.MicroscopeQuality
+import net.aiscope.gdd_app.model.Sample
+import net.aiscope.gdd_app.model.SamplePreparation
+import net.aiscope.gdd_app.repository.MicroscopeQualityDto
+import net.aiscope.gdd_app.repository.SampleMetadataDto
+import net.aiscope.gdd_app.repository.SamplePreparationDto
+
+fun Sample.toDto() = SampleDto(
+    id = id,
+    healthFacility = healthFacility,
+    microscopist = microscopist,
+    disease = disease ?: "",
+    preparation = preparation?.toDto(),
+    microscopeQuality = microscopeQuality?.toDto(),
```

Comment:
Might be a good idea to add a test to see if the selected preparation & microscope show up in the JSON that is saved with the sample in the end

Reply:
Definitely! Done, thanks!